### ISSUE TITLE
`@property` inconsistently documented

### ISSUE THREAD

**Issue by rpitasky** [mitmproxy/pdoc#574]
#### Problem Description
Our codebase uses [a lot of `@property`](https://github.com/TI-Toolkit/tivars_lib_py/blob/fa070d04d0cb8767feeb3273c70fc9776c7c87c1/tivars/var.py#L352-L435) and we're seeing that [only some of these are documented](https://ti-toolkit.github.io/tivars_lib_py/tivars/var.html#TIEntry.data_length). 

Most curious to me is the fact that `TIEntry.data_length` is documented while the other `@property`s are not.

#### Steps to reproduce the behavior:
1. Save the following minimal code to `bug.py`
```
class Bug:
    @property
    def a(self):
        return "a"
```
2. Run pdoc on it (I use `python3 -m pdoc bug.py` but [our CI](https://github.com/TI-Toolkit/tivars_lib_py/blob/fa070d04d0cb8767feeb3273c70fc9776c7c87c1/.github/workflows/docs.yml#L28) does it normally, but I doubt this matters)
3. Observe that the `@property` code is not documented.

#### System Information
Paste the output of "pdoc --version" here.
pdoc 13.1.1, python 3.10

**Comment by rpitasky** [comment]
It looks like the issue is that the @properties don't have docstrings; I find this behavior rather curious.

What is the justification for the default being to not document these?

**Comment by mhils** [comment]
See https://github.com/mitmproxy/pdoc/issues/280 and https://github.com/mitmproxy/pdoc/issues/411 - that should give you some background. I guess I hold no super strong opinions here. :-)

**Comment by kg583** [comment]
I don't have strong opinions about the choice of intended behavior either, but there is still inconsistency.

According to #280, "pdoc documents all attributes that have either a docstring *or* a type annotation." (emphasis mine), but properties are only being documented if there is a docstring *and* type annotation. Indeed, [this property has both](https://github.com/TI-Toolkit/tivars_lib_py/blob/fa070d04d0cb8767feeb3273c70fc9776c7c87c1/tivars/var.py#L353) while [this property is only annotated](https://github.com/TI-Toolkit/tivars_lib_py/blob/fa070d04d0cb8767feeb3273c70fc9776c7c87c1/tivars/var.py#L430), and only former [appears in the docs](https://ti-toolkit.github.io/tivars_lib_py/tivars/var.html#TIEntry.data_length).

**Comment by mhils** [comment]
#280 is older than #411, see https://github.com/mhils/pdoc/commit/a6f21be24582e981ca1a60f24a031dd316f709f0

**Comment by rpitasky** [comment]
Thank you for your rapid responses. Well, that's certainly a conundrum- you have users with diametrically opposing positions here! I guess my opinion coming into this was that (by default) *everything* should be documented. I personally prefer this because it shows developers explicitly what still needs documentation, and at least half of the utility of code documentation for users is to show what exists.

Perhaps this whole issue could be resolved through additional documentation in pdoc's Quickstart. This could be easily be roped into the opening beats; I propose editing the Quickstart example to include something with neither type annotations or a docstring and changing

> If you look closely, you'll notice that docstrings are interpreted as Markdown. For example, \`pdoc\` is rendered as `pdoc`﻿. Additionally, identifiers such as the type annotation for `Dog.friends` are automatically linked.

into
> If you look closely, you'll notice that docstrings are interpreted as Markdown. For example, \`pdoc\` is rendered as `pdoc`﻿. Additionally, identifiers such as the type annotation for `Dog.friends` are automatically linked. Items without type annotations or docstrings are omitted by default, but they ccan be included by &lt;insert-way-to-change-this-default-here-if-it-exists>.

(or a derivative of the above that doesn't dramatically alter the flow of the whole section), with the additional subtleties being noted under the ["How can I document variables"](https://pdoc.dev/docs/pdoc.html#document-variables) section. I think this is a fundamental enough design choice that it deserves a prominent spot on that page.

**Comment by kg583** [comment]
> #280 is older than #411, see [mhils@a6f21be](https://github.com/mhils/pdoc/commit/a6f21be24582e981ca1a60f24a031dd316f709f0)

Ah, I didn't quite pick that out of the thread.

It does nonetheless feel a bit weird for the case of properties, since although they act like variables in syntactic respects, being a `@property` communicates the fact that this is a value the user could want to fetch, in contrast with an instance variable which might user-facing or might not be (if not made explicitly private). And although there _should_ be a docstring, presumably one could have a property with a sufficiently self-documenting name and return type.

**Comment by mhils** [comment]
Thank you two! I think there's a strong argument that variables shouldn't behave any other from functions or classes in terms of default visibility. I've just pushed #575, which makes it so that variables are documented by default - irrespective of whether they have a docstring or not. I've linked this issue in the changelog - if someone comes up with a convincing use case for why this is particularly annoying, we might add a `--documented-only` flag or something like that. Let's see! 😃
